### ISSUE TITLE
The explanation of joint and independent learning in Table 5.

### ISSUE THREAD

**Issue by Paranioar** [Paranioar/SGRAF#18]
In the class EncoderSimilarity(nn.Module) of [model.py](https://github.com/Paranioar/SGRAF/blob/python3.6/model.py),
we define the SGR and SAF modules separately:
```
if module_name == 'SGR':
    self.SGR_module = nn.ModuleList([GraphReasoning(sim_dim) for i in range(sgr_step)])
elif module_name == 'SAF'.
    self.SAF_module = AttentionFiltration(sim_dim)
else.
    raise ValueError('Invalid input of opt.module_name in opts.py')
```
In other words, you can only train either SGR or SAF respectively, and average these two similarities offline to get the results of **independent learning** in Table5.

For **joint learning**, you only need to define both SGR and SAF simultaneously as follows:
```
self.SGR_module = nn.ModuleList([GraphReasoning(sim_dim) for i in range(sgr_step)])
self.SAF_module = AttentionFiltration(sim_dim)
```
Activating both modules in the *forward* process and averaging their similarities online to get the final result, which is then fed into the Rank Loss function.
We experimentally found that there is no gain in this joint learning, as shown in Table 5.
